src/test/java/vampireNumbersTests/VampireNumbersTest.java

```diff
@@ -0,0 +1,56 @@
+package vampireNumbersTests;
+
+import com.tngtech.java.junit.dataprovider.*;
+import org.junit.*;
+import org.junit.runner.*;
+import vampireNumbers.Permutation;
+import vampireNumbers.VampireNumbersDefinition;
+
+import java.util.*;
+
+import static org.hamcrest.MatcherAssert.*;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(DataProviderRunner.class)
+public class VampireNumbersTest {
+    private VampireNumbersDefinition definition = new VampireNumbersDefinition();
+
+    @DataProvider
+    public static Object[][] numbersToDefine() {
+        return new Object[][]{
+                {new char[]{'1', '2', '6', '0'}, 1260, true},
+                {new char[]{'1', '5', '5', '6'}, 1556, false},
+                {new char[]{'9', '9', '9', '9'}, 9999, false},
+                {new char[]{'6', '8', '8', '0'}, 6880, true},
+        };
+    }
+
+    @Test
+    @UseDataProvider("numbersToDefine")
+    public void defineIfVampire(char[] c, int number, boolean result) {
+        assertThat(definition.isVampire(c, number), is(result));
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void exceptionThrown_OnSmallNumber() {
+        definition.isVampire(new char[]{'2'}, 2);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void exceptionThrown_OnLargeNumber() {
+        definition.isVampire(new char[]{'1', '0', '0', '0', '0'}, 10000);
+    }
+
+    @Test
+    public void permutationTest() {
+        Permutation.permute(new char[]{'1', '2', '6', '0'});
+        Set<String> result = Permutation.getResult();
+        Iterator<String> iterator = result.iterator();
+        int numberInResult = 0;
+        while (iterator.hasNext()) {
+            iterator.next();
+            numberInResult++;
+        }
+        assertThat(numberInResult, is(24));
```

Comment:
if it failed how could you reproduce it?

Reply:
Assertion is simplified in 2663d0cf91b298f16ae6d1ab661bee2d6300463b.